arbnode/batch_poster.go

```diff
@@ -148,10 +154,13 @@ type BatchPosterDangerousConfig struct {
 type BatchPosterConfig struct {
 	Enable                             bool `koanf:"enable"`
 	DisableDapFallbackStoreDataOnChain bool `koanf:"disable-dap-fallback-store-data-on-chain" reload:"hot"`
-	// Max batch size.
-	MaxSize int `koanf:"max-size" reload:"hot"`
+	// Maximum calldata batch size for EthDA.
+	MaxCalldataBatchSize int `koanf:"max-calldata-batch-size" reload:"hot"`
```

Comment:
let's support both. If user sets max-size we should post a warning that it's deprecated and set max-calldata-batch-size.

Reply:
I'm in favor of changing configuration names if the name is clearer/more consistent as long as we include it in the release notes. For the more complex nested configuration structs maintaining backwards compatibility is hard, but it's not too hard to add it for this one if you really think we should. I think I would have it print an ERROR instead of a WARNing since warnings tend to get ignored.

Reply:
@joshuacolvin0 what's your take on that?